DIsability VA Nepal.R

```diff
@@ -27,58 +40,143 @@ xtabs(data = pilotfinal, ~six9_os, addNA = TRUE)
+  # JK: general pattern for gather always has 3 elements: field, value, range of values you want to put in a single column
+  # JK: I normally use the actual word field, though modify it in cases like this
+  gather(fieldeye, value, six60_os:phsix9_od) %>%
+  separate(fieldeye, into=c("field", "eye"), sep="_") %>%
+  spread(field, value, convert=TRUE) %>%
+  mutate(visionscreen=case_when(six60==0 & phsix60==0 ~ 1, 
+                                six60==0 & phsix60==1 & phsix18==0 ~ 2, 
+                                six60==0 & phsix60==1 & phsix18==1 & phsix9==0 ~ 3, 
+                                six60==0 & phsix60==1 & phsix18==1 & phsix9==1 ~ 4,
+                                six60==1 & six18==0 & phsix18==0 ~ 5,
+                                six60==1 & six18==0 & phsix18==1 & phsix9==0 ~ 6,
+                                six60==1 & six18==0 & phsix18==1 & phsix9==1 ~ 7,
+                                six60==1 & six18==1 & six9==0 & phsix9==0 ~ 8,
+                                six60==1 & six18==1 & six9==0 & phsix9==1 ~ 9,
+                                six60==1 & six18==1 & six9==1 ~ 10,
+                                TRUE ~ NA_real_)) %>%
+  gather(fielddis, value, dis_see:dis_comm) %>%
+  separate(fielddis, into=c("field", "distype"), sep="_") %>%
+  spread(field, value, convert=TRUE) %>%
+  rename(dis_score=dis) %>%
+  mutate(dis_score= recode_factor(dis_score,  `1`="no", `2`="some", `3`="alot", `4`="unable", .ordered=TRUE))
+  
   
-#change disability to factorial
-disvaODfact <- disvaOD %>%
- mutate(dis_see_fact=as.factor(disvaOD$dis_see),
-  #check that what I did worked
-  #xtabs(data=disvaODfact, ~dis_see + dis_see_fact, addNA=TRUE)
-  #class(disvaODfact$dis_see_fact), it did so convert the rest of the variables to factorial
-  dis_hear_fact=as.factor(disvaOD$dis_hear),
-  dis_walk_fact=as.factor(disvaOD$dis_walk),
-  dis_rem_fact=as.factor(disvaOD$dis_rem),
-  dis_wash_fact=as.factor(disvaOD$dis_wash),
-  dis_comm_fact=as.factor(disvaOD$dis_comm)) 
 
-#labelling the levels of the above factors
-disfactlabels <- c("no", "some", "alot", "unable")
-levels(disvaODfact$dis_see_fact) = disfactlabels  #check my work with levels(disvaODfact$dis_see_fact)
-levels(disvaODfact$dis_hear_fact) = disfactlabels
-levels(disvaODfact$dis_walk_fact) = disfactlabels
-levels(disvaODfact$dis_rem_fact) = disfactlabels
-levels(disvaODfact$dis_wash_fact) = disfactlabels
-levels(disvaODfact$dis_comm_fact) = disfactlabels 
+,
+dis_hear=recode_factor(dis_hear, `1`="no", `2`="some", `3`="alot", `4`="unable", .ordered=TRUE),
+dis_walk=recode_factor(dis_walk, `1`="no", `2`="some", `3`="alot", `4`="unable", .ordered=TRUE),
+dis_rem= recode_factor(dis_rem,  `1`="no", `2`="some", `3`="alot", `4`="unable", .ordered=TRUE),
+dis_wash=recode_factor(dis_wash, `1`="no", `2`="some", `3`="alot", `4`="unable", .ordered=TRUE),
+dis_comm=recode_factor(dis_comm, `1`="no", `2`="some", `3`="alot", `4`="unable", .ordered=TRUE),
```

Comment:
instead of using the as.factor() argument you can use recode_factor(colname, specify level names, .ordered=TRUE to create an ordered factor)